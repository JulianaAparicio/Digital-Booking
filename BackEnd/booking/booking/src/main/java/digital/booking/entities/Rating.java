package digital.booking.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "ratings")
public class Rating {

    @Id
    @SequenceGenerator(name="rating_seg",initialValue = 1,allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE,generator = "rating_seq")

    @NotNull
    private Long id;

    @Column
    private Integer score;

    @ManyToMany
    @JoinColumn(name = "users",nullable = false)
    private List<User> users;

}
