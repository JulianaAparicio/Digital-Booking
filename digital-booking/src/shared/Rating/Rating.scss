.db-rating {
    display: inline-flex;
    padding: 0 5px;
    svg {
        margin: 0 -5px;
    }
}