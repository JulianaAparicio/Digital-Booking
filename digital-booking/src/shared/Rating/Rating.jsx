import { useState } from "react";
import StartFilledIcon from "../Icons/StarFilledIcon";
import './Rating.scss';

export default function Rating({rate}) {
    const [rating, setRating] = useState(new Array(rate).fill(1))

    return (
        <div className="db-rating">
            {
                rating.map((el) => <StartFilledIcon/>)
            }
            
        </div>
        
    )
}