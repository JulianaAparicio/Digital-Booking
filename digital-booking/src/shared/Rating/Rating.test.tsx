import {describe, test, expect, beforeEach} from 'vitest';
import {render, screen} from '@testing-library/react';
import React from 'react';
import Rating from './Rating';

describe('Rating test', () => {
    let component;

    beforeEach(() => {
        component = render(<Rating rate={4} />)
    })

    test('Should render the component', () => {
        expect(component).toMatchSnapshot();
    })

    test('Should render four starts icon', () => {
        expect(component.container.firstChild.childNodes.length).toBe(4);
    })

    test('Should has the class db-rating', () => {
        expect(component.container.firstElementChild?.classList.contains('db-rating')).toBeTruthy();
    })
})