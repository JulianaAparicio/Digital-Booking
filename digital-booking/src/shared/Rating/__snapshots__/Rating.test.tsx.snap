// Vitest Snapshot v1

exports[`Rating test > Should render the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="db-rating"
      >
        <svg
          fill="none"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
            fill="#1DBEB4"
          />
        </svg>
        <svg
          fill="none"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
            fill="#1DBEB4"
          />
        </svg>
        <svg
          fill="none"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
            fill="#1DBEB4"
          />
        </svg>
        <svg
          fill="none"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
            fill="#1DBEB4"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="db-rating"
    >
      <svg
        fill="none"
        height="13"
        viewBox="0 0 15 13"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
          fill="#1DBEB4"
        />
      </svg>
      <svg
        fill="none"
        height="13"
        viewBox="0 0 15 13"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
          fill="#1DBEB4"
        />
      </svg>
      <svg
        fill="none"
        height="13"
        viewBox="0 0 15 13"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
          fill="#1DBEB4"
        />
      </svg>
      <svg
        fill="none"
        height="13"
        viewBox="0 0 15 13"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.27778 0L8.91174 4.83688H14.1994L9.92159 7.82624L11.5555 12.6631L7.27778 9.67376L3.00001 12.6631L4.63397 7.82624L0.3562 4.83688H5.64382L7.27778 0Z"
          fill="#1DBEB4"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
